src/mds/events/EMetaBlob.h

```diff
@@ -207,12 +206,14 @@ class EMetaBlob {
   private:
     mutable bufferlist dnbl;
     mutable bool dn_decoded;
-    mutable list<std::shared_ptr<fullbit> > dfull;
+    mutable list<fullbit> dfull;
```

Comment:
Also looks like this could have been a vector?

Reply:
Also, does it (still) need to be `mutable`?

Reply:
mutable is required by EMetaBlob::dirlump::_encode_bits/_decode_bits

Reply:
dremote and dnull can be vector, but dfull can not be. because fullbit is large,  I delete its copy constructer